### PR TITLE
Added LM75 binding

### PR DESCRIPTION
**Pull request by ZhangGaoxing**
Digital Temperature Sensor LM75

### CODE REVIEW

File: src/devices/Lm75/Lm75.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+using System.Device.I2c;
+
+namespace Iot.Device.Lm75
+{
+    /// <summary>
+    /// Digital Temperature Sensor LM75
+    /// </summary>
+    public class Lm75 : IDisposable
+    {
+        I2cDevice _sensor;
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: explicit private

File: src/devices/Lm75/samples/Program.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+namespace Iot.Device.Lm75.Samples
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            I2cConnectionSettings settings = new I2cConnectionSettings(1, Lm75.I2cAddress);
+            UnixI2cDevice device = new UnixI2cDevice(settings);
+
+            using(Lm75 sensor=new Lm75(device))
```
Comment (krwq) [inline src/devices/Lm75/samples/Program.cs]: spaces around `=`

File: src/devices/Lm75/samples/Program.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (krwq) [inline src/devices/Lm75/samples/Program.cs]: license header

File: src/devices/Lm75/Lm75.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        /// <summary>
+        /// LM75 Temperature (℃)
+        /// </summary>
+        public double Temperature { get => GetTemperature(); }
+
+        private bool _shutdown;
+        /// <summary>
+        /// LM75 Shutdown
+        /// </summary>
+        public bool Shutdown { get { return _shutdown; } set { SetShutdown(value); _shutdown = value; } }
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: I would go with some word with past tense i.e. Disabled. Shutdown sounds like a function name

Comment (ZhangGaoxing) [reply src/devices/Lm75/Lm75.cs]: Aha, the datasheet says shutdown. I will change the name.😄

File: src/devices/Lm75/Lm75.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            double temp = 0;
+            ushort raw = (ushort)((readBuff[0] << 3) | (readBuff[1] >> 5));
+            if ((readBuff[0] & 0x80) == 0)
+            {
+                temp = raw * 0.125;
+            }
+            else
+            {
+                // two's complement
+                raw |= 0xF800;
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: isn't two's complement be equivalent to using BinaryPrimitives.ReadLittle/BigEndian?

File: src/devices/Lm75/Register.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device.Lm75
+{
+    public enum Register : byte
```
Comment (krwq) [inline src/devices/Lm75/Register.cs]: this should be private

File: src/devices/Lm75/Lm75.cs (hunk 5 above)
Comment (ZhangGaoxing) [reply src/devices/Lm75/Lm75.cs]: Aha, I don't quite understand what you mean. `using BinaryPrimitives.ReadInt16Little/BigEndian` need to pass in a Span<T>. This is a binary complement for negative numbers. Temperature register data contains undefined bits, and BinaryPrimitives.ReadInt16Little/BigEndian cannot be used directly😄. @krwq

Comment (krwq) [reply src/devices/Lm75/Lm75.cs]: @ZhangGaoxing I'm fine with this as is for now but I suspect if you just cast raw to short you might get same result - not sure what the `raw |= 0xF800` does though - I'd have to dig more to check it so fine to leave as is if you get correct results

File: src/devices/Lm75/Lm75.cs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                // temperature < 0
+                // two's complement
+                raw |= 0xF800;
+                raw = (ushort)(~raw + 1);
+
+                temp = raw * (-1) * 0.125;
+            }
+
+            return Math.Round(temp, 1);
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: why are you rounding?

File: src/devices/Lm75/Lm75.cs [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        /// <param name="isShutdown">Shutdown when value is true.</param>
+        private void SetShutdown(bool isShutdown)
+        {
+            _sensor.WriteByte((byte)Register.LM_CONFIG);
+            byte config = _sensor.ReadByte();
+
+            config &= 0xFE;
+            if (isShutdown)
+                config |= 0x01;
+            else
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: else can be removed completely `config |= 0x00` is a nop, isn't it?

File: src/devices/Lm75/Lm75.cs (hunk 7 above)
Comment (ZhangGaoxing) [reply src/devices/Lm75/Lm75.cs]: The temperature of this sensor is accurate to 0.1

File: src/devices/Lm75/Lm75.cs (hunk 8 above)
Comment (ZhangGaoxing) [reply src/devices/Lm75/Lm75.cs]: Yes, I just want to keep the code readable. I'll keep this operation as a comment.

File: src/devices/Lm75/Lm75.cs (hunk 5 above)
Comment (ZhangGaoxing) [reply src/devices/Lm75/Lm75.cs]: `0xF800` just sets the high five unused bits

File: src/devices/Lm75/Lm75.cs [hunk 9]
```diff
@@ -89,8 +89,11 @@ private void SetShutdown(bool isShutdown)
             config &= 0xFE;
             if (isShutdown)
                 config |= 0x01;
-            else
-                config |= 0x00;
+            //else
```
Comment (krwq) [inline src/devices/Lm75/Lm75.cs]: remove the commented out code 😄

**Pull request merged**